op-service/metrics/cli.go

```diff
@@ -32,21 +35,44 @@ func CLIFlags(envPrefix string) []cli.Flag {
 			Usage:   "Enable the metrics server",
 			EnvVars: opservice.PrefixEnvVar(envPrefix, "METRICS_ENABLED"),
 		},
-		&cli.StringFlag{
+		&cli.GenericFlag{
 			Name:    ListenAddrFlagName,
 			Usage:   "Metrics listening address",
-			Value:   defaultListenAddr, // TODO(CLI-4159): Switch to 127.0.0.1
+			Value:   NewFlagValue(defaultListenAddr), // TODO(CLI-4159): Switch to 127.0.0.1
 			EnvVars: opservice.PrefixEnvVar(envPrefix, "METRICS_ADDR"),
 		},
-		&cli.IntFlag{
+		&cli.GenericFlag{
 			Name:    PortFlagName,
 			Usage:   "Metrics listening port",
-			Value:   defaultListenPort,
+			Value:   NewFlagValue(fmt.Sprint(defaultListenPort)),
 			EnvVars: opservice.PrefixEnvVar(envPrefix, "METRICS_PORT"),
 		},
 	}
 }
 
+type FlagValue string
+
+func NewFlagValue(s string) *FlagValue {
+	return (*FlagValue)(&s)
+}
+
+func (fv *FlagValue) Set(value string) error {
+	value = strings.ToLower(value) // ignore case
+	*fv = FlagValue(value)
+	return nil
+}
+
+func (fv FlagValue) String() string {
+	return string(fv)
+}
+
+func (fv *FlagValue) Clone() any {
+	cpy := *fv
+	return &cpy
+}
+
+var _ cliapp.CloneableGeneric = (*FlagValue)(nil)
+
```

Comment:
These can be `StringFlag` and `IntFlag`. The CI issue with flags earlier was a problem with `GenericFlag`, which has been resolved in `develop` for any existing generic flags.

If you like we can implement specialized host and port flag types with specialized value parsing, based on the `GenericFlag` type, but it's not required for the main cleanup of this PR: deduplicating the metrics docs functionality.

Reply:
You may need to rebase on `develop` to get the CI to pass though

Reply:
Very well, I'll revert them to `StringFlag` and `IntFlag` for now but would be open to doing that in a subsequent PR.